pycqed/analysis_v2/Two_state_T1_analysis.py

```diff
@@ -0,0 +1,333 @@
+        self.raw_data_dict['time units'] = a.sweep_unit[0]
+
+        self.raw_data_dict['value_names'] = a.value_names
+        self.raw_data_dict['value_units'] = a.value_units
+        self.raw_data_dict['measurementstring'] = a.measurementstring
+        self.raw_data_dict['timestamp_string'] = a.timestamp_string
+
+        self.raw_data_dict['cal_pts_zero'] = OrderedDict()
+        self.raw_data_dict['cal_pts_one'] = OrderedDict()
+        self.raw_data_dict['cal_pts_two'] = OrderedDict()
+        self.raw_data_dict['measured_values_I'] = OrderedDict()
+        self.raw_data_dict['measured_values_X'] = OrderedDict()
+
+        for i, val_name in enumerate(a.value_names):
+            self.raw_data_dict['cal_pts_zero'][val_name] = \
+                a.measured_values[i][-6:-4]
+            self.raw_data_dict['cal_pts_one'][val_name] = \
+                a.measured_values[i][-4:-2]
+            self.raw_data_dict['cal_pts_two'][val_name] = \
+                a.measured_values[i][-2:]
+            self.raw_data_dict['measured_values_I'][val_name] = \
+                a.measured_values[i][::2][:-3]
+            self.raw_data_dict['measured_values_X'][val_name] = \
+                a.measured_values[i][1::2][:-3]
+
+        self.raw_data_dict['folder'] = a.folder
+        self.raw_data_dict['timestamps'] = self.timestamps
+        a.finish()  # closes data file
+
+    def process_data(self):
+        self.proc_data_dict = deepcopy(self.raw_data_dict)
+
+        for key in ['V0', 'V1', 'V2', 'SI', 'SX', 'P0', 'P1', 'P2', 'M_inv']:
+            self.proc_data_dict[key] = OrderedDict()
+
+        for val_name in self.raw_data_dict['value_names']:
+            V0 = np.mean(
+                self.raw_data_dict['cal_pts_zero'][val_name])
+            V1 = np.mean(
+                self.raw_data_dict['cal_pts_one'][val_name])
+            V2 = np.mean(
+                self.raw_data_dict['cal_pts_two'][val_name])
+
+            self.proc_data_dict['V0'][val_name] = V0
+            self.proc_data_dict['V1'][val_name] = V1
+            self.proc_data_dict['V2'][val_name] = V2
+
+            SI = self.raw_data_dict['measured_values_I'][val_name]
+            SX = self.raw_data_dict['measured_values_X'][val_name]
+
+            self.proc_data_dict['SI'][val_name] = SI
+            self.proc_data_dict['SX'][val_name] = SX
+
+            P0, P1, P2, M_inv = populations_using_rate_equations(
+                SI, SX, V0, V1, V2)
+            self.proc_data_dict['P0'][val_name] = P0
+            self.proc_data_dict['P1'][val_name] = P1
+            self.proc_data_dict['P2'][val_name] = P2
+            self.proc_data_dict['M_inv'][val_name] = M_inv
+
+    def run_fitting(self):
+        super().run_fitting()
+        self.fit_res['fit_res_P2'] = OrderedDict()
+        self.fit_res['fit_res_P1'] = OrderedDict()
+        self.fit_res['fit_res_P0'] = OrderedDict()
+        decay_mod = lmfit.Model(ExpDecayFunclocal, independent_vars='t')
```

Comment:
Why is there a local copy of this (`ExpDecayFunclocal`) function? Is there anything different between this one and the one in fit_mods?

Reply:
Just for my convenience but I have changed. Now code is using fitting_model module.